connection-loader/src/main/java/com/terracotta/connection/api/AbstractConnectionService.java

```diff
@@ -48,6 +50,23 @@ public final Connection connect(URI uri, Properties properties) throws Connectio
     return internalConnect(clientConfig);
   }
 
+  private void validateURI(URI uri) throws ConnectionException {
+    String path = uri.getPath();
+    if (path != null && path.length() > 0 && !path.equals("/")) {
```

Comment:
The path can't be null in a URI. And aren't you looking for `path.contains('/')`? I'm not sure what this code is for. Are we validating an URI like `terracotta://example.com:9540/my-application`?

Reply:
A path can be null (see https://docs.oracle.com/javase/8/docs/api/java/net/URI.html#getPath-- ).

We are trying to ensure that no-one specifies a path. So "terracotta://example.com:9540" and "terracotta://example.com:9540/" are valid - although the second one is dubious if you ask me.

Reply:
Stupid me... I thought the path was the full URL. Now I get it. In fact you don't need the length then no? Could be `if(path != null && !path.equals("/"))`

Reply:
@ChrisGreenaway : URIs such as `news:terracotta:9410` or `mailto:toto@terracotta:9410` will pass the validation and have null path. Is it normal ?

Reply:
@mathieucarbou I don't think we should go crazy about it. The intent is, I think, so prevent paths, not to entirely validate the URI.